### ISSUE TITLE
A11y Accessibility issues with SlickGrid

### ISSUE THREAD

**Issue by shrhegde** [6pac/SlickGrid#678]
We are using SlickGrid with our Angular JS application and there are below issues with Accessibility:

SlickGrid version: 2.3.17
AngularJS: 1.8.x
Operating System: Win11
Chromium Edge: Version 98.0.1108.13 (Official build) dev (64-bit)
Chrome: Version 96.0.4664.110 (Official Build) (64-bit)
Screen Reader: Narrator, NVDA(2021.3.1)

1. A11y: While navigating with keyboard, expansion of column width in the grid is not accessible.
2. A11y: In high contrast mode there is no visible focus in the grid while navigating through keyboard. 
3. A11y: Screenreader does not announce the table name for grid present in the page page when it is on focus.
4. A11y: Screenreader is not announcing the grid data along with column headers in the grid.
5. A11y: Narrator is not announcing any information about the grid present on the screen when it is on focus.
6.  A11y: When navigating inside the grid with keyboard, it is getting trapped inside the grid and there is no way to come out of the grid.
7. A11y: When pagination is enabled on the grid, The Bulb icon present in the right corner of the grid, next/previous buttons, are not keyboard accessible. 
8. A11y: When pagination is enabled on the grid, the icons >> ,<<, bulb icon etc. all such icons are not visible properly.
![image](https://user-images.githubusercontent.com/86716990/168240012-93f51b22-cf20-487a-89b1-359adc7ab68d.png)

9. A11y: When pagination is enabled on the grid, The Name property is not provided for Bulb icon button present in the right bottom corner.
![image](https://user-images.githubusercontent.com/86716990/168239934-aabd9644-723e-46f1-8948-efeca1d1647b.png)

**Comment by 6pac** [comment]
hi @shrhegde , 

Yes, accessibility has been a problem with the grid for a while. The main problem is that I'm not at all an expert in accessibilty and it appears to be a quite complex area. I am also aware that these things can evolve quite rapidly, so I'm also concerned that I might make changes in line with an older standard that is not really the most active one at the present day.  See https://github.com/6pac/SlickGrid/issues/206 for a rundown of my concerns.

I note there are a number of accessibility issues/PRs still open from the last few years.

I'm more than happy to make changes to the grid, but I feel that I need to collaborate with someone who really knows their way around web accessibility and can make reliably correct calls.

I note you are using the A11y tools. Are you just a user of the tools, or do you have a good knowledge of the issues around accessible web design?

Ben Mc (@6pac)

**Comment by ghiscoding** [comment]
> @6pac Can I get some timeline on when these bugs will be fixed as these bugs are affecting the accessibility of our application

@shrhegde 
Wow I'm just amaze to see such comment in an Open Source project, seriously why don't you just contribute if you want it faster? Please realize that we are all doing this work for free in our spare time, we certainly don't work for you, and getting more contributions would be a lot more helpful.

**Comment by ghiscoding** [comment]
That's too bad that you decided to simply close the issue instead of contributing to the project which could benefit everyone including yourself, but it is what it is.

**Comment by ghiscoding** [comment]
closed by #764 to be released in next major v4.0
